orion/pkg/controllers/achieve_controller_test.go

```diff
@@ -51,40 +51,83 @@ func TestGetAllAchievements_Success(t *testing.T) {
 }
 
 //
-// Test Get Achieve
+// Test Get Published
 //
-func TestGetAchievement_Success(t *testing.T) {
-	achieveService.mockGetById = func(id uint) (domains.Achieve, error) {
-		achieve := createMockAchievement(1, 2020, "message1")
-		return achieve, nil
+func TestGetPublishedAchievements_Success(t *testing.T) {
+	achieveService.mockGetAll = func(publishedOnly bool) ([]domains.Achieve, error) {
+		return []domains.Achieve{
+			createMockAchievement(
+				1,
+				2020,
+				"message1",
+			),
+			createMockAchievement(
+				2,
+				2021,
+				"message2",
+			),
+		}, nil
 	}
 	services.AchieveService = &achieveService
 
 	// Create new HTTP request to endpoint
-	recorder := sendHttpRequest(t, http.MethodGet, "/api/achievements/v1/achievement/1", nil)
+	recorder := sendHttpRequest(t, http.MethodGet, "/api/achievements/v1/all?published=true", nil)
 
 	// Validate results
 	assert.EqualValues(t, http.StatusOK, recorder.Code)
-	var achieve domains.Achieve
-	if err := json.Unmarshal(recorder.Body.Bytes(), &achieve); err != nil {
+	var achieves []domains.Achieve
+	if err := json.Unmarshal(recorder.Body.Bytes(), &achieves); err != nil {
 		t.Errorf("unexpected error: %v\n", err)
 	}
-	assert.EqualValues(t, 1, achieve.Id)
-	assert.EqualValues(t, 2020, achieve.Year)
-	assert.EqualValues(t, "message1", achieve.Message)
+	assert.EqualValues(t, 1, achieves[0].Id)
+	assert.EqualValues(t, 2020, achieves[0].Year)
+	assert.EqualValues(t, "message1", achieves[0].Message)
+
+	assert.EqualValues(t, 2, achieves[1].Id)
+	assert.EqualValues(t, 2021, achieves[1].Year)
+	assert.EqualValues(t, "message2", achieves[1].Message)
+
+	assert.EqualValues(t, 2, len(achieves))
 }
 
-func TestGetAchievement_Failure(t *testing.T) {
-	achieveService.mockGetById = func(id uint) (domains.Achieve, error) {
-		return domains.Achieve{}, errors.New("not found")
+//
+// Test Get Unpublished
+//
+func TestGetUnpublishedAchievements_Success(t *testing.T) {
```

Comment:
I would say move this to the publish_controller_test.go. Yes, we are creating mock achievements and returning mocked achievements but effectively we are testing the publish_controller endpoint.
In the near future, you can include other services to mock other data points as well.